src/xiaomi-ble/parser.py

```diff
@@ -0,0 +1,95 @@
+"""Parser for Xiaomi BLE advertisements.
+
+This file is shamelessly copied from the following repository:
+https://github.com/Ernst79/bleparser/blob/c42ae922e1abed2720c7fac993777e1bd59c0c93/package/bleparser/xiaomi.py
+
+MIT License applies.
+"""
+from __future__ import annotations
+
+import logging
+from struct import unpack
+
+from bluetooth_sensor_state_data import BluetoothData
+from home_assistant_bluetooth import BluetoothServiceInfo
+from sensor_state_data import SensorLibrary
+
+_LOGGER = logging.getLogger(__name__)
+
+BBQ_LENGTH_TO_TYPE = {
+    12: ("iBBQ-1", "<h"),
+    14: ("iBBQ-2", "<HH"),
+    18: ("iBBQ-4", "<hhhh"),
+    22: ("iBBQ-6", "<hhhhhh"),
+}
+
+# ToDo Change Inkbrid parser to Xiaomi
+XIAOMI_NAMES = {
+    "sps": "IBS-TH",
+    "tps": "IBS-TH2/P01B",
+}
+
+
+def convert_temperature(temp: float) -> float:
+    """Temperature converter."""
+    if temp > 0:
+        return temp / 10.0
+    return 0
+
+
+def short_address(address: str) -> str:
+    """Convert a Bluetooth address to a short address."""
+    results = address.replace("-", ":").split(":")
+    if len(results[-1]) == 2:
+        return f"{results[-2].upper()}{results[-1].upper()}"
+    return results[-1].upper()
+
+
+class XIAOMIBluetoothDeviceData(BluetoothData):
+    """Date update for XIAOMI Bluetooth devices."""
+
+    def _start_update(self, service_info: BluetoothServiceInfo) -> None:
+        """Update from BLE advertisement data."""
+        _LOGGER.debug("Parsing Xiaomi BLE advertisement data: %s", service_info)
+        manufacturer_data = service_info.manufacturer_data
+        if not manufacturer_data:
+            return
+        last_id = list(manufacturer_data)[-1]
+        data = int(last_id).to_bytes(2, byteorder="little") + manufacturer_data[last_id]
+        self.set_device_manufacturer("XIAOMI")
+        self._process_update(service_info.name, service_info.address, data)
+
+    def _process_update(self, local_name: str, address: str, data: bytes) -> None:
+        """Update from BLE advertisement data."""
+        _LOGGER.debug("Parsing XIAOMI BLE advertisement data: %s", data)
+        msg_length = len(data)
+
+        if (device_type := XIAOMI_NAMES.get(local_name)) and msg_length == 9:
+            self.set_device_type(device_type)
+            self.set_device_name(f"{device_type} {short_address(address)}")
+            (temp, hum) = unpack("<hH", data[0:4])
+            bat = int.from_bytes(data[7:8], "little")
+            if local_name == "sps":
+                self.update_predefined_sensor(SensorLibrary.TEMPERATURE, temp / 100)
```

Comment:
I'm going to change all the names on the sensorlibrary to include the unit type as well since some sensors report different units and it's confusing this is always in a single unit.

Reply:
Once you have the first commit merge I will update the repo as I make the breaking changes so you don't have to figure out what's shifting in the foundation

Reply:
No problem, we will update according to your changes, I don't expect the Xiaomi sensors to make it in the next release anyways. 

A side note about temperature units. A special sensor is the Xiaomi LYWSD03MMC, which can be flashed with ATC firmware (main advantage is that it can report 10X per second, in stead of 1x per 10 minutes. But with this ATC firmware, it starts sending data in Fahrenheit, if you set the display to Fahrenheit, and in Celsius if you set it to Celsius. In BLE monitor, I therefore had to make an option where the user could change the Unit of measurement the sensors sends its data. But that is for later (ATC is a different parser), with Xiaomi firmware it is always Celsius, regardless of the display setting. But it might be something to keep in mind.